test/FunctionsV2/ProtobufUtilsTests.cs

```diff
@@ -0,0 +1,389 @@
+            Assert.Equal(expectedVersion, startOrchestrationResult.Version);
+        }
+
+        /// <summary>
+        /// Tests that ToEntityBatchResult properly passes the default version to all actions.
+        /// </summary>
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public void ToEntityBatchResult_PassesDefaultVersionToActions()
+        {
+            // Arrange
+            var startNewOrchestrationAction = new P.StartNewOrchestrationAction
+            {
+                Name = "TestOrchestrator",
+                InstanceId = "test-instance-id",
+            };
+
+            var operationAction = new P.OperationAction
+            {
+                StartNewOrchestration = startNewOrchestrationAction,
+            };
+
+            var entityBatchResult = new P.EntityBatchResult
+            {
+                EntityState = "{}",
+            };
+            entityBatchResult.Actions.Add(operationAction);
+            entityBatchResult.Results.Add(new P.OperationResult
+            {
+                Success = new P.OperationResultSuccess
+                {
+                    Result = "null",
+                },
+            });
+
+            var defaultVersion = "2025-10-23";
+
+            // Act
+            var result = entityBatchResult.ToEntityBatchResult(defaultVersion);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Single(result.Actions);
+            var action = Assert.IsType<StartNewOrchestrationOperationAction>(result.Actions[0]);
+            Assert.Equal(defaultVersion, action.Version);
+        }
+
+        /// <summary>
+        /// Tests that SendSignalOperationAction is not affected by the default version parameter.
+        /// </summary>
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public void ToOperationAction_SendSignal_NotAffectedByDefaultVersion()
+        {
+            // Arrange
+            var sendSignalAction = new P.SendSignalAction
+            {
+                Name = "TestOperation",
+                InstanceId = "@entity@test",
+                Input = "\"test-input\"",
+            };
+
+            var operationAction = new P.OperationAction
+            {
+                SendSignal = sendSignalAction,
+            };
+
+            var defaultVersion = "2025-10-23";
+
+            // Act
+            var result = operationAction.ToOperationAction(defaultVersion);
+
+            // Assert
+            Assert.NotNull(result);
+            var sendSignalResult = Assert.IsType<SendSignalOperationAction>(result);
+            Assert.Equal("TestOperation", sendSignalResult.Name);
+            Assert.Equal("@entity@test", sendSignalResult.InstanceId);
+        }
+
+        /// <summary>
+        /// Tests backward compatibility - when no default version is provided, behavior should match previous implementation.
```

Comment:
Nit: the notion of "backward compatibility" and "previous implementation" will become outdated soon after this PR is merged. Consider removing it from the comment and the test name.